lib/galaxy/workflow/modules.py

```diff
@@ -81,6 +81,9 @@ def get_type( self ):
     def get_name( self ):
         return self.name
 
+    def get_version( self ):
+        return None
```

Comment:
There are some refactoring in here that look pretty nice to me.

But this one feels like more refactoring where a desire to treat everything in the frontend like "a tool" is causing poor design decisions on the backend. Version was a concept that I feel like only applied to tools - "Input Dataset" modules shouldn't have a ``get_version`` in the abstract unless that is something we plan to use - unless they have an inherent version. 

Same with ``get_post_job_actions`` - it seems inappropriate that every module has a list of post job actions when only one kind of module even produces jobs.